--- a/config.tests/unix/slog2/slog2.cpp	2025-02-06 11:57:20.115176350 -0500
+++ b/config.tests/unix/slog2/slog2.cpp	2025-02-06 11:57:27.416255443 -0500
@@ -31,7 +31,7 @@
 **
 ****************************************************************************/
 
-#include <slog2.h>
+#include <sys/slog2.h>
 
 int main(int, char **)
 {

--- a/src/corelib/global/qlogging.cpp	2025-02-06 16:13:51.586158185 -0500
+++ b/src/corelib/global/qlogging.cpp	2025-02-06 16:13:57.391217093 -0500
@@ -53,7 +53,7 @@
 #include <qt_windows.h>
 #endif
 #ifdef QT_USE_SLOG2
-#include <slog2.h>
+#include <sys/slog2.h>
 #endif
 
 #ifdef Q_OS_ANDROID
