--- a/src/plugins/platforms/qnx/qqnxintegration.cpp	2025-02-24 11:57:27.119921001 -0500
+++ b/src/plugins/platforms/qnx/qqnxintegration.cpp	2025-02-24 11:58:08.188371171 -0500
@@ -120,16 +120,9 @@
         options |= QQnxIntegration::AlwaysFlushScreenContext;
     }
 
-// On Blackberry the first window is treated as a root window
-#ifdef Q_OS_BLACKBERRY
     if (!paramList.contains(QLatin1String("no-rootwindow"))) {
-        options |= QQnxIntegration::RootWindow;
-    }
-#else
-    if (paramList.contains(QLatin1String("rootwindow"))) {
         options |= QQnxIntegration::RootWindow;
     }
-#endif
     return options;
 }
 
